sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/models/TimeInterval.java

```diff
@@ -0,0 +1,264 @@
+        }
+
+        if (duration != null) {
+            return duration.toString();
+        }
+        // should not reach here
+        throw LOGGER.logExceptionAsError(new IllegalStateException("Invalid date time range"));
+    }
+
+    /**
+     * This method takes an ISO 8601 formatted time interval string and returns an instance of {@link TimeInterval}.
+     * @param value The ISO 8601 formatted time interval string.
+     * @return An instance of {@link TimeInterval}.
+     * @throws IllegalArgumentException if {@code value} is not in the correct format.
+     */
+    public static TimeInterval parse(String value) {
+        Objects.requireNonNull(value);
+
+        String[] parts = value.split("/");
+        if (parts.length == 1) {
+            // duration
+            Duration duration = parseDuration(parts[0]);
+            if (duration == null || parts[0].length() + 1 == value.length()) {
+                // input strings like "PT24H/" are invalid
+                throw LOGGER.logExceptionAsError(
+                        new IllegalArgumentException(String.format(ERROR_MESSAGE, value)));
+            }
+
+            return new TimeInterval(duration);
+        }
+
+        if (parts.length == 2) {
+            Duration startDuration = parseDuration(parts[0]);
+            OffsetDateTime startTime = parseTime(parts[0]);
+
+            Duration endDuration = parseDuration(parts[1]);
+            OffsetDateTime endTime = parseTime(parts[1]);
+
+            if (startDuration != null && endTime != null) {
+                return new TimeInterval(startDuration, endTime);
+            }
+            if (startTime != null && endTime != null) {
+                return new TimeInterval(startTime, endTime);
+            }
+            if (startTime != null && endDuration != null) {
+                return new TimeInterval(startTime, endDuration);
+            }
+        }
+        throw LOGGER.logExceptionAsError(
+                new IllegalArgumentException(String.format(ERROR_MESSAGE, value)));
+    }
+
+    @Override
+    public String toString() {
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("[");
+        if (duration != null) {
+            return sb.append("duration: ").append(duration).append("]").toString();
+        }
+
+        if (startTime != null) {
+            sb.append("start time: ").append(startTime);
+        }
+        if (startDuration != null) {
+            sb.append("start duration: ").append(startDuration);
+        }
+        sb.append(", ");
+        if (endTime != null) {
+            sb.append("end time: ").append(endTime);
+        }
+        if (endDuration != null) {
+            sb.append("end duration: ").append(endDuration);
+        }
+        sb.append("]");
+        return sb.toString();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
```

Comment:
You could simplify this with using `instanceof`

Reply:
Yeah, in this case, we could potentially use `instanceof` because the class is marked `final`. But I prefer to use `getClass()` because if A extends B, then `a instanceof b` is true and not vice versa. 

See - https://youtrack.jetbrains.com/issue/IDEA-7692